src/utils/registerNewNetwork.ts

```diff
@@ -74,38 +74,75 @@ export const registerNewNetwork = async (
     blockTime: arbitrumSdkConstants.ARB_MINIMUM_BLOCK_TIME_IN_SECONDS,
   };
 
-  let isL1NetworkRegistered = false;
-  try {
-    await getL1Network(l1Provider);
-    isL1NetworkRegistered = true;
-  } catch {
-    isL1NetworkRegistered = false;
-  }
+  // If the parent chain is an Arbitrum chain, we need to register parent and child chain both as L2 chains
+  if (await isArbitrumChain(parentProvider)) {
+    const isParentNetworkRegistered = await isNetworkRegistered(parentProvider, { type: 'L2' });
+    const parentNetwork: Omit<L2Network, 'tokenBridge'> = {
+      chainID: parentNetworkInfo.chainId,
+      name: parentNetworkInfo.name,
+      confirmPeriodBlocks: (await rollup.confirmPeriodBlocks()).toNumber(),
+      ethBridge: {
+        bridge: await rollup.bridge(),
+        inbox: await rollup.inbox(),
+        outbox: await rollup.outbox(),
+        rollup: rollup.address,
+        sequencerInbox: await rollup.sequencerInbox(),
+      },
+      explorerUrl: '',
+      isArbitrum: true,
+      isCustom: true,
+      partnerChainIDs: [childNetworkInfo.chainId],
+      retryableLifetimeSeconds: 7 * 24 * 60 * 60,
+      partnerChainID: 1,
```

Comment:
We don't have any information about the L1 parent of the L2 chain here, default to Mainnet